src/main/java/uk/gov/hmcts/reform/sendletter/services/LetterService.java

```diff
@@ -36,13 +38,32 @@ public UUID send(LetterRequest letter, String serviceName) {
 
         final String messageId = generateChecksum(letter);
 
-        log.info("Generated message: id = {} for letter with print queue id = {}", messageId, letter.type);
+        log.info("Generated message: id = {}", messageId);
 
-        byte[] pdf = pdfCreator.create(letter);
-        Letter dbLetter = new Letter(messageId, serviceName, null, letter.type, pdf);
+        Optional<Letter> result = letterRepository.findByMessageIdAndStateOrderByCreatedAtDesc(
+            messageId,
+            LetterState.Created
+        );
+
+        return result
+            .map(l -> {
+                UUID id = l.getId();
+                log.info("Same message found already created. Returning {} letter", id);
```

Comment:
is 'Returning 8eb874b2-e975-4006-aa2c-9883fa66b065 letter' the message you want to log?